File: backend/app/routers/health.py
from datetime import datetime, timezone

from fastapi import APIRouter

router = APIRouter(tags=["health"])


@router.get("/healthz")
def read_health() -> dict[str, str]:
    return {
        "status": "ok",
        "service": "cerul-api",
        "timestamp": datetime.now(timezone.utc).isoformat(),
    }

File: api/src/routes/health.ts
import { Hono } from "hono";

export function createHealthRouter(): any {
  const router = new Hono();

  router.get("/healthz", (c: any) =>
    c.json({
      status: "ok",
      service: "cerul-api",
      timestamp: new Date().toISOString()
    })
  );

  router.get("/", (c: any) => {
    const config = c.get("config");
    return c.json({
      name: "cerul-api",
      status: "ok",
      environment: config.environment
    });
  });

  router.get("/v1/meta", (c: any) => {
    const config = c.get("config");
    return c.json({
      service: "cerul-api",
      framework: "hono",
      environment: config.environment
    });
  });

  return router;
}
